pkg/generate/app/cmd/newapp.go

```diff
@@ -913,6 +915,48 @@ func (c *AppConfig) Run() (*AppResult, error) {
 	}, nil
 }
 
+// crossStreamCircularTagReference inherits some logic from imageapi.FollowTagReference, but differs in that a) it is only concerned
+// with whether we can definitively say the IST chain is circular, and b) can cross image stream boundaries;
+// not in imageapi pkg (see imports above) like other helpers cause of import cycle with the image client
+func (c *AppConfig) crossStreamCircularTagReference(stream *imageapi.ImageStream, tag string) bool {
+	seen := sets.NewString()
+	// get our own copy
+	strm, err := c.ImageClient.ImageStreams(stream.ObjectMeta.Namespace).Get(stream.ObjectMeta.Name, metav1.GetOptions{})
+	if strm == nil || err != nil {
+		return false
+	}
+	for {
+		if seen.Has(strm.ObjectMeta.Namespace + ":" + strm.ObjectMeta.Name + ":" + tag) {
+			// circular reference
+			return true
+		}
+		seen.Insert(strm.ObjectMeta.Namespace + ":" + strm.ObjectMeta.Name + ":" + tag)
+		tagRef, ok := strm.Spec.Tags[tag]
+		if !ok {
+			// no tag at the end of the rainbow
+			return false
+		}
+		if tagRef.From == nil || tagRef.From.Kind != "ImageStreamTag" {
+			// terminating tag
+			return false
+		}
+		if strings.Contains(tagRef.From.Name, ":") {
```

Comment:
is this not mandatory for a tag ref of kind ImageStreamTag ?

Reply:
i guess based on the logic below if it doesn't contain a colon it's assumed that From.Name refers to another tag within this imagestream.  ok.

Reply:
yep that is what it is